src/index.ts

```diff
@@ -213,21 +102,44 @@ export class OperationContext {
 	}
 
 	/**
-	 * Creates a new stack entry in the operation. Ideally, call this when calling a new
-	 * asynchronous operation to track its context separately while attached to the high-level
-	 * operation.
+	 * Sets one or multiple values on the current context. If the keys already
+	 * exist, they will be overwritten.
+	 * @param values additional values to append
 	 */
-	next(): OperationContextEntry {
+	setValues(values: Record<string, any>): OperationContext {
 		if (this.timeoutError) {
 			throw this.timeoutError
 		}
 		if (!this.isRunning()) {
-			throw this.createError(`Cannot continue a ${this.status} operation`)
+			throw this.createError(`Cannot set values on a ${this.status} operation`)
 		}
+		this.stack.push({ values, error: new Error('---') })
```

Comment:
I know this is dumb, but is there value in `'---'` over `''`?  Those 3 bytes count for something 😂